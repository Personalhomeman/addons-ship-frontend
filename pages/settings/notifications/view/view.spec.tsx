import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import View, { Props } from '.';

describe('NotificationSettings View', () => {
  const defaultProps: Props = {
    onAddEmail: jest.fn(),
    appContacts: [],
    onNotificationPreferenceChanged: jest.fn()
  };

  it('renders correctly', () => {
    const tree = shallowToJson(shallow(<View {...defaultProps} />));

    expect(tree).toMatchSnapshot();
  });

  it('has an enabled save button', () => {
    const tree = shallowToJson(shallow(<View {...defaultProps} onSave={jest.fn()} />));

    expect(tree).toMatchSnapshot();
  });
});
