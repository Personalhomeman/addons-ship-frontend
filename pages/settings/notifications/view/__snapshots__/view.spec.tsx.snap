// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettings View has an enabled save button 1`] = `
<Base
  paddingVertical="x8"
>
  <Text
    paddingVertical="x3"
    size="x5"
    weight="bold"
  >
    Email Notifications
  </Text>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    paddingVertical="x6"
  >
    <_default
      onAddEmail={[MockFunction]}
    />
  </Base>
  <Base
    margin="x4"
  >
    <_default
      appContacts={Array []}
      onNotificationPreferenceChanged={[MockFunction]}
    />
  </Base>
  <Flex
    alignChildrenHorizontal="end"
    direction="horizontal"
    margin="x12"
  >
    <Buttons
      gap="x2"
    >
      <Button
        level="secondary"
        size="medium"
      >
        Cancel
      </Button>
      <Button
        disabled={false}
        level="primary"
        onClick={[MockFunction]}
        size="medium"
      >
        Save
      </Button>
    </Buttons>
  </Flex>
</Base>
`;

exports[`NotificationSettings View renders correctly 1`] = `
<Base
  paddingVertical="x8"
>
  <Text
    paddingVertical="x3"
    size="x5"
    weight="bold"
  >
    Email Notifications
  </Text>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    paddingVertical="x6"
  >
    <_default
      onAddEmail={[MockFunction]}
    />
  </Base>
  <Base
    margin="x4"
  >
    <_default
      appContacts={Array []}
      onNotificationPreferenceChanged={[MockFunction]}
    />
  </Base>
  <Flex
    alignChildrenHorizontal="end"
    direction="horizontal"
    margin="x12"
  >
    <Buttons
      gap="x2"
    >
      <Button
        level="secondary"
        size="medium"
      >
        Cancel
      </Button>
      <Button
        disabled={true}
        level="primary"
        size="medium"
      >
        Save
      </Button>
    </Buttons>
  </Flex>
</Base>
`;
